### PR TITLE
Update c layer doc

### LINKED ISSUE #1162: fatal error: 'gtk/gtk.h' file not found
## Autocomplete with gtk files

Example:
#include<gtk/gtk.h>

int main(int argc, char **argv) {
  GtkWidget *win;
  gtk_init(&argc, &argv);
  win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
  gtk_window_set_title(GTK_WINDOW(win), "Hello there");
  g_signal_connect(win, "destroy", G_CALLBACK(gtk_main_quit), NULL);
  gtk_widget_show(win);
  gtk_main();

}

I created the file '.clang_complete' and still continue with the error gtk/gtk.h not found!

Below the '.clang_complete' file contents:
-pthread
-I/usr/include/gtk-3.0
-I/usr/include/at-spi2-atk/2.0
-I/usr/include/at-spi-2.0
-I/usr/include/dbus-1.0
-I/usr/lib/x86_64-linux-gnu/dbus-1.0/include
-I/usr/include/gtk-3.0
-I/usr/include/gio-unix-2.0/
-I/usr/include/mirclient
-I/usr/include/mircore
-I/usr/include/mircookie
-I/usr/include/cairo
-I/usr/include/pango-1.0
-I/usr/include/harfbuzz
-I/usr/include/pango-1.0
-I/usr/include/atk-1.0
-I/usr/include/cairo
-I/usr/include/pixman-1
-I/usr/include/freetype2
-I/usr/include/libpng12
-I/usr/include/gdk-pixbuf-2.0
-I/usr/include/libpng12
-I/usr/include/glib-2.0
-I/usr/lib/x86_64-linux-gnu/glib-2.0/include
-lgtk-3
-lgdk-3
-lpangocairo-1.0
-lpango-1.0
-latk-1.0
-lcairo-gobject
-lcairo
-lgdk_pixbuf-2.0
-lgio-2.0
-lgobject-2.0
-lglib-2.0

## Environment Information
- OS: Distributor ID: Ubuntu
Description:    Ubuntu 16.04.3 LTS
Release:        16.04
Codename:       xenial

- vim version: VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Nov 24 2016 16:44:48)
captura de tela de 2017-12-22 14-19-59

### PR DESCRIPTION
**Pull request by wsdjeg**
close #1162 
ref: https://github.com/neomake/neomake/issues/972
In spacevim, we use `.clang` to control the argv for c project, these argvs should be applyed to neomake and completion_plugin.

also the spacevim's build-in runner.

cc @blueyed

### CODE REVIEW

File: autoload/SpaceVim/autocmds.vim [hunk 1]
```diff
@@ -159,6 +159,7 @@ function! SpaceVim#autocmds#VimEnter() abort
     call SpaceVim#layers#core#tabline#def_colors()
     set showtabline=2
   endif
+  call SpaceVim#plugins#projectmanager#RootchandgeCallback()
```
Comment (blueyed) [inline autoload/SpaceVim/autocmds.vim:162]: s/chandge/changed/ ?

File: autoload/SpaceVim/layers/checkers.vim [hunk 2]
```diff
@@ -8,7 +8,7 @@ let s:STRING = SpaceVim#api#import('data#string')
 function! SpaceVim#layers#checkers#plugins() abort
   let plugins = []
 
-  if g:spacevim_enable_neomake
+  if g:spacevim_enable_neomake && g:spacevim_enable_ale == 0
     call add(plugins, ['neomake/neomake', {'merged' : 0, 'loadconf' : 1 , 'loadconf_before' : 1}])
   elseif g:spacevim_enable_ale
```
Comment (blueyed) [inline autoload/SpaceVim/layers/checkers.vim:13]: I can imagine allowing to enable both Neomake and ALE (even though ALE might complain by default).
I am lazy-loading ALE sometimes after Neomake, just to test / see how it works by now.
I think having one of them doing autolinting, but calling the other manually (e.g. automaking with Neomake, but using ALE's fixer) makes sense.

File: autoload/SpaceVim/layers/lang/c.vim [hunk 3]
```diff
@@ -81,3 +94,49 @@ function! s:language_specified_mappings() abort
...
+
+function! s:update_clang_flag() abort
+  if filereadable('.clang')
+    let argvs = readfile('.clang')
+    call s:update_checkers_argv(argvs, ['c', 'cpp'])
+    call s:update_autocomplete_argv(argvs, ['c', 'cpp'])
+  endif
+endfunction
+
+if g:spacevim_enable_neomake && g:spacevim_enable_ale == 0
```
Comment (blueyed) [inline autoload/SpaceVim/layers/lang/c.vim:106]: Same as above: should not check for `g:spacevim_enable_ale` here maybe?!

File: autoload/SpaceVim/layers/lang/c.vim [hunk 4]
```diff
@@ -81,3 +94,49 @@ function! s:language_specified_mappings() abort
...
+            \ '%f:%l: %m'
+            \ }
+    endfor
+  endfunction
+elseif g:spacevim_enable_ale
+  function! s:update_checkers_argv(argv, fts) abort
+    " g:ale_c_clang_options
+    for ft in a:fts
+      let g:ale_{ft}_clang_options = ' -fsyntax-only -Wall -Wextra -I./ ' . join(a:argv, ' ')
+      let g:ale_{ft}_clang_executabl = s:clang_executable
```
Comment (blueyed) [inline autoload/SpaceVim/layers/lang/c.vim:130]: s/executabl/executable/ ?

File: autoload/SpaceVim/autocmds.vim (hunk 1 above)
Comment (wsdjeg) [reply autoload/SpaceVim/autocmds.vim:162]: sorry，it has been merged， But I will fix this type, thank your for reviewing code.

File: autoload/SpaceVim/layers/checkers.vim (hunk 2 above)
Comment (wsdjeg) [reply autoload/SpaceVim/layers/checkers.vim:13]: I still prefer use one of them by defaule. users can use custom plugins to add ale and ale's config.
but in spacevim, if neomake is enabled, ale will be disabled, and this config will not be applyed to ale. but users still can use spacevim's custom plugin option to add ale.

SpaceVim contains too many plugins now. even though most of them are lazzy loaded. but I user do not want to see plugins with same function are installed at same time.

**Pull request merged**